packages/terminal/src/node/shell-process.ts

```diff
@@ -85,7 +68,7 @@ export class ShellProcess extends TerminalProcess {
                 cols: options.cols || ShellProcess.defaultCols,
                 rows: options.rows || ShellProcess.defaultRows,
                 cwd: getRootPath(options.rootURI),
-                env: setUpEnvVariables(options.env),
```

Comment:
I think we could keep that behavior but re-use `mergeProcessEnv`. This would prevent making such a breaking change here. You can extract the `mergeProcessEnv` method into a regular function somewhere in order to use it in both places.

Reply:
Done